test/app-luatest/gh_7808_rel_paths_in_diag_and_debug_info_test.lua

```diff
@@ -0,0 +1,25 @@
+local bit = require('bit')
+local misc = require('test.luatest_helpers.misc')
+local netbox = require('net.box')
+local t = require('luatest')
+
+local g = t.group()
+
+g.test_rel_paths_in_diag = function(_)
+    local _, err = pcall(function() netbox.self:call('') end)
+    t.assert_not(err.trace[1].file:startswith('/'))
+end
+
+g.test_rel_paths_in_debug_info = function(_)
+    misc.skip_if_macOS()
+
+    local readelf = ("readelf --debug-dump=info %s"):format(arg[-1])
+    local grep_comp_dir = "grep DW_AT_comp_dir"
+    local awk = "awk '{print $9}'"
+    local grep_abs_path = "grep '^/'"
+    local rc = os.execute(table.concat({readelf, grep_comp_dir, awk, grep_abs_path}, '|'))
+    -- `os.execute` behaves like `system`.
+    rc = bit.rshift(bit.band(rc, 0xff00), 8)
+    -- `grep` returns 1 if no lines were selected.
+    t.assert_equals(rc, 1)
+end
```

Comment:
Several suggestions:

1. If you need to run the test only on Linux, let's test for 'on linux' instead of 'not on mac os'. Tarantool works on FreeBSD and NetBSD, for example.
2. [jit.os](https://luajit.org/ext_jit.html#jit_os) provides simpler OS check and you'll not need a helper. If you'll update the helpers anyway, notify Yaroslav Lobankov: he is working on moving them to luatest itself (see https://github.com/tarantool/luatest/pull/247).
3. I would check for `readelf` existence (say, run `readelf --version`) and skip the test if is not exists.
4. I would use Lua's string functions instead of grep/awk for better portability.
5. You may be insterested in tarantool's built-in `popen` module or `io.popen`.

Reply:
Points 1-3: applied your third suggestion and simply test for `readline` availability.

Points 4-5: rewrote the test to parse using `io.open` (`popen` does not have convenient functions for parsing) and `string.gmatch` — unfortunately, the test now takes 14 seconds compared to negligible in the `os.execute` version.

Reply:
Thanks a lot for your suggestions!